cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkRuntimeContextProvider.java

```diff
@@ -209,7 +209,7 @@ private static synchronized SparkRuntimeContext createIfNotExists() {
       logAppenderInitializer.initialize();
 
       // For spark running natively on k8s, we may need to initialize the TokenManager for internal identity.
-      if (SecurityUtil.isInternalAuthEnabled(cConf)) {
+      if (clusterMode == ClusterMode.ON_PREMISE && SecurityUtil.isInternalAuthEnabled(cConf)) {
```

Comment:
I'm not really clear on why we need this, is the problem with tethered runs? Do they go through a different code path for auth?

Reply:
With existing logic, ON_PREM runtime is treated as a cluster-native app which initializes its own internal token generator, whereas the ISOLATED mode uses a saved runtime token instead. Ultimately we'll want to unify this logic (since ON_PREM should still be treated as executing untrusted user-provided code), but there are additional considerations for design, including how to handle token renewal, token scope reduction, etc.

Reply:
The reason this flow was breaking is because the Guice modules for TokenManager were not being included when running in ISOLATED mode.